scripts/image-saver.py

```diff
@@ -1,85 +1,179 @@
-        file. If file_name is blank (an empty string), it will default to
-        camera-timestamp.
+    camera: pcdsdevices.areadetector.detectors.PCDSAreaDetector
+        The actual camera object containing the hdf51 plugin used to
+        collect and save images. Created by build_cam in cam_load.
+    num_images: int
+        The number of images to take.
+    path: string
+        Path where the hdf5 file containing the images should be saved to.
+    filename: string
+        Name of hdf5 file images are saved in. If empty, will default to
+        {camera}-{timestamp}. If not empty, timestamp will be appended
+        to ensure all captures have unique filenames.
     """
-    current_name = args.filename
+    # Default filename is camera-timestamp
+    if not filename:
+        filename = f'{camera.name}-{int(time())}'
+    # Setting up hdf51 plugin
+    camera.hdf51.file_name.put(filename)
+    camera.hdf51.file_template.put('%s%s_%03d.h5')
+    camera.hdf51.num_capture.put(num_images)
+    camera.hdf51.auto_save.put(True)
+    camera.hdf51.file_write_mode.put('Capture')
+    camera.hdf51.enable.put('Enabled')
+    camera.hdf51.capture.put('Capture')
+    # Creating callbacks to know when collecting/saving images is complete
+    collecting = True
+    writing = True
+
+    def _collecting_cb(value, **kwargs):
+        """Get a callback for collecting changes"""
+        nonlocal collecting
+        collecting = value
+
+    def _writing_cb(value, **kwargs):
+        """Get a callback when writing changes"""
+        nonlocal writing
+        writing = value
+    # Start collecting and saving
+    collect_uid = camera.hdf51.capture.subscribe(_collecting_cb)
+    write_uid = camera.hdf51.write_file.subscribe(_writing_cb)
+    # Clean up subscribers and disable plugin when finished
+    while collecting and camera.hdf51.num_captured.get() < num_images:
+        sleep(0.1)
+    camera.hdf51.capture.unsubscribe(collect_uid)
+    while writing:
+        sleep(0.1)
+    camera.hdf51.write_file.unsubscribe(write_uid)
+    camera.hdf51.enable.put('Disabled')
+
+
+def load_camera(hutch, cam_name):
+    """
+    Creates camera object.
+
+    Gets camera details based on name and hutch-specific
+    camviewer config file and uses build_cam to create an
+    individual camera object. Will exit if the config file
+    or the camera cannot be found.
+
+    Parameters
+    ----------
+    hutch: string
+        Name of hutch the camera is in, needed to find and
+        load the correct config file.
+    cam_name: string
+        Name of camera to load.
+
+    Returns
+    -------
+    camera: pcdsdevices.areadetector.detectors.PCDSAreaDetector
+        Camera object created by build_cam with hdf51 plugin
+        member to take and save images.
+    """
+    # Get cam_load functions
+    syspath.append(f'/reg/g/pcds/pyps/apps/hutch-python/{hutch}'
+                   '/dev/hutch-python/hutch_python/cam_load.py')
```

Comment:
This directory is not guaranteed to exist, it should be expected that the dev folder will be periodically cleared as fixes propagate to tagged libraries.

If the script is ran using the normal central conda environments (`source /cds/group/pcds/pyps/conda/pcds_conda`) or from the hutch python envs (`source /cds/group/pcds/pyps/apps/hutch-python/{hutch}/{hutchenv}`) then `hutch_python.cam_load` should be importable without any path appending.

`/reg/g` and `/cds/group` are network mounts to the same disk, where `/reg/g` is the old/deprecated path.

Reply:
Good to know, can I assume that this will be run in either of those environments or should I check for/enforce that somehow?

Reply:
I think there are two sensible approaches:
- "This is a python script that requires `pcds_conda`, please source `pcds_conda` before running this."
- We can write a shell script that sources pcds_conda and then runs this python script